include/maliput_osm/builder/road_network_builder.h

```diff
@@ -0,0 +1,139 @@
+///
+/// Example of use:
+/// @code{cpp}
+/// const std::map<std::string, std::string> builder_configuration {
+///   {"maliput_osm::builder::kRoadGeometryId", "appian_way_road_geometry"},
+///   {"maliput_osm::builder::kOsmPath", "appian_way.osm"},
+///   {"maliput_osm::builder::kOrigin", "{34., 58.}"},
+///   {"maliput_osm::builder::kLinearTolerance", "1e-3"},
+///   {"maliput_osm::builder::kAngularTolerance", "1e-3"},
+/// };
+/// auto road_network = maliput_osm::loader::Load<maliput_osm::builder::RoadNetworkBuilder>(builder_configuration);
+/// @endcode
+///
+/// @{
+
+/// A string that works as ID of the RoadGeometry.
+///   - Default: @e "maliput"
+static constexpr char const* kRoadGeometryId{"road_geometry_id"};
+
+/// Path to the OSM file to be loaded.
+///   - Default: ""
+static constexpr char const* kOsmFile{"osm_file"};
+
+/// Lat/lon coordinate of the origin of the OSM file.
+/// The format of the 2-dimensional vector that is expected to be passed
+/// should be {X, Y}. Same format as maliput::math::Vector2 is
+/// serialized.
+///   - Default: @e "{0., 0.}"
+static constexpr char const* kOrigin{"origin"};
+
+/// RoadGeometry's linear tolerance.
+///   - Default: @e "5e-2"
+static constexpr char const* kLinearTolerance{"linear_tolerance"};
+
+/// RoadGeometry's angular tolerance.
+///   - Default: @e "1e-3"
+static constexpr char const* kAngularTolerance{"angular_tolerance"};
+
+/// RoadGeometry's scale length.
+///   - Default: @e "1.0"
+static constexpr char const* kScaleLength{"scale_length"};
+
+/// Translation from maliput to maliput_osm inertial frame.
+/// The format of the 3-dimensional vector that is expected to be passed
+/// should be {X, Y, Z}. Same format as maliput::math::Vector3 is
+/// serialized.
+///   - Default: @e "{0., 0., 0.}"
+static constexpr char const* kInertialToBackendFrameTranslation{"inertial_to_backend_frame_translation"};
+
+/// Path to the configuration file to load a RoadRulebook
+///   - Default: ""
+static constexpr char const* kRoadRuleBook{"road_rule_book"};
+
+/// Path to the configuration file to load a RoadRulebook
+///   - Default: ""
+static constexpr char const* kRuleRegistry{"rule_registry"};
+
+/// Path to the configuration file to load a TrafficLightBook
+///   - Default: ""
+static constexpr char const* kTrafficLightBook{"traffic_light_book"};
+
+/// Path to the configuration file to load a PhaseRingBook
+///   - Default: ""
+static constexpr char const* kPhaseRingBook{"phase_ring_book"};
+
+/// Path to the configuration file to load a IntersectionBook
+///   - Default: ""
+static constexpr char const* kIntersectionBook{"intersection_book"};
+
+/// @}
+
+}  // namespace config
+
+class RoadNetworkBuilder {
+ public:
+  MALIPUT_NO_COPY_NO_MOVE_NO_ASSIGN(RoadNetworkBuilder);
+
+  /// Constructs a RoadNetworkBuilder.
+  ///
+  /// @param builder_config Builder configuration.
+  explicit RoadNetworkBuilder(const std::map<std::string, std::string>& builder_config)
```

Comment:
unordered_map?

Reply:
Yeah `unordered_map` should be ok. I used `map` because we are currently using `map` for the RoadNetworkLoader plugin (which under the hood forwards the configuration to the RoadNetworkBuilder.). And that would require a conversion (no cast) from one to another at the plugin implementaiton. (THis would affect this https://github.com/maliput/maliput_osm/pull/15 )

Would that be ok?

Reply:
Let's keep it for now as is.